server/fleet/service.go

```diff
@@ -538,8 +538,9 @@ type Service interface {
 
 	// ListHostUpcomingActivities lists the upcoming activities for the specified
 	// host. Those are activities that are queued or scheduled to run on the host
-	// but haven't run yet.
-	ListHostUpcomingActivities(ctx context.Context, hostID uint, opt ListOptions) ([]*Activity, *PaginationMetadata, error)
+	// but haven't run yet. It also returns the total (unpaginated) count of upcoming
+	// activities.
+	ListHostUpcomingActivities(ctx context.Context, hostID uint, opt ListOptions) (uint, []*Activity, *PaginationMetadata, error)
```

Comment:
Thinking out loud, how would you feel about making the `count` part of the `*PaginationMetadata`, but have it not being rendered (`json:"-"`)? It is related to pagination, and it feels a bit weird to expand this to return the count separately. Eventually we could backport other places where we get the count along a page of data to use this?

Reply:
I mean, in the future we could even update our API so that pagination metadata holds the `count`, instead of having it as a separate top-level field, but that's certainly too big a change for now/this PR, just something we could work towards.

Reply:
@mna, I was thinking along those lines too :) 

Do you think we should simply add it to the existing type definition or define a new type with embedding? If we just add to the exiting type, is it worth the trouble of using `*uint` to differentiate the zero value in places where the count is not included by design?

```
type PaginationMetadata struct {
	HasNextResults        bool `json:"has_next_results"`
	HasPreviousResults bool `json:"has_previous_results"`
        TotalResults              *uint `json:"-"`
}
```

Reply:
@gillespi314 I think that since we don't automatically render it (where we use and want it, we have to copy it explicitly from the `PaginationMetadata` to the `Counts` field in the `response` struct), it should be fine to have it defined as `uint` (not pointer).

Reply:
@gillespi314 re your other point about embedding, I'd avoid that and add it to the existing struct, again we don't render it in the `PaginationMetadata` so it's really about collecting the pagination-related metadata in the same struct, and (AFAICT) shouldn't affect existing uses that don't support the counts.